lib/Service/FontService.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * @copyright Copyright (c) 2022 Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @author Julien Veyssier <eneiluj@posteo.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Richdocuments\Service;
+
+
+use OCA\Richdocuments\AppInfo\Application;
+use OCP\Files\IAppData;
+use OCP\Files\NotFoundException;
+use OCP\Files\SimpleFS\ISimpleFile;
+use OCP\Files\SimpleFS\ISimpleFolder;
+use OCP\ICacheFactory;
+use OCP\IConfig;
+use OCP\IURLGenerator;
+
+class FontService {
+	private const INVALIDATE_FONT_LIST_CACHE_AFTER_SECONDS = 3600;
+
+	/**
+	 * @var IAppData
+	 */
+	private $appData;
+	/**
+	 * @var \OCP\ICache
+	 */
+	private $cache;
+	/**
+	 * @var IURLGenerator
+	 */
+	private $url;
+	/**
+	 * @var IConfig
+	 */
+	private $config;
+
+	public function __construct(IAppData $appData,
+								ICacheFactory $cacheFactory,
+								IURLGenerator $url,
+								IConfig $config) {
+		$this->appData = $appData;
+		$this->cache = $cacheFactory->createDistributed(Application::APPNAME);
+		$this->url = $url;
+		$this->config = $config;
+	}
+
+	/**
+	 * @return ISimpleFolder
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	private function getFontAppDataDir(): ISimpleFolder {
+		try {
+			return $this->appData->getFolder('fonts');
+		} catch (NotFoundException $e) {
+			return $this->appData->newFolder('fonts');
+		}
+	}
+
+	/**
+	 * @return ISimpleFolder
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	private function getFontOverviewAppDataDir(): ISimpleFolder {
+		try {
+			return $this->appData->getFolder('font-overviews');
+		} catch (NotFoundException $e) {
+			return $this->appData->newFolder('font-overviews');
+		}
+	}
+
+	/**
+	 * Get the list of available font files
+	 *
+	 * @return array
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function getFontFiles(): array {
+		$cacheKey = 'fontFiles';
```

Comment:
This cache key seems to get never invalidated, causing the font file list to not update

Reply:
That's weird. Nothing fancy here. We set the ttl when setting the cache value. Usual stuff. Are you sure it never gets invalidated? Default cache ttl is one hour.

Anyway let's get rid of caching here.